### PR TITLE
Create view to add categories

### LINKED ISSUE #73: Create a view to add Categories
**Is your feature request related to a problem? Please describe.**
We need a new view in the admin dashboard to add categories.

**Describe the solution you'd like**
Should develop a view under `Home/scene` as `AddCategory` to be functioned to `POST` a category into the database.

**Additional context**
Related thread: https://sef.discourse.group/t/updating-the-academix-admin-panel/131

### PR DESCRIPTION
**Pull request by Gravewalker666**
## Purpose

The purpose of this PR is to fix #73 

## Goals
* To create a new scene under `Home/scenes` to add new categories

## Approach
* Created a new folder under `Home/scenes`
* Created `index.tsx` and `interfaces.ts` in that folder
* Created a from using `AntD Form` component 
* Wired the above component with the `POST category` endpoint on `sef-core/backend` using Axios.
* Added buttons above the categories table and in the `Add` menu to access the created form.

## Screenshot
image
image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
* Java Version: OpenJDK 11.0.7
* TestServer: Tomcat 9
* Os: MacOs Catalina 10.15.3
* TestBrowser: Google Chrome

## Learning
* https://ant.design/components/overview/

### CODE REVIEW

File: frontend/app/src/scenes/Home/index.tsx [hunk 1]
```diff
@@ -94,15 +94,22 @@ class Home extends React.Component<HomeProps> {
...
-                  <Link to="/dashboard/universities/add/">University</Link>
-                </Menu.Item>
-                <Menu.Item>
-                  <Link to="/dashboard/ambassadors/add">Ambassador</Link>
-                </Menu.Item>
-                <Menu.Item>
-                  <Link to="/dashboard/academix/item/add">Item</Link>
-                </Menu.Item>
+                <Menu.ItemGroup title="Multiverse">
+                  <Menu.Item>
```
Comment (jayasanka-sack) [inline frontend/app/src/scenes/Home/index.tsx:98]: Add unique keys for each menu item

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    super(props);
+    this.state = {
+      isLoading: false,
+    };
+  }
+
+  onFinish = (values: any) => {
+    this.setState({ isLoading: true });
+    axios
+      .post(`${window.origin}/core/academix/admin/categories`, values)
```
Comment (jayasanka-sack) [inline frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:35]: let's use window.location.origin

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      .then((res: AxiosResponse<Category>) => {
+        if (res.status == 201) {
+          this.setState({ isLoading: false });
+          notification.success({
+            message: 'Success!',
+            description: 'Successfully Created a Category',
+          });
+          this.props.history.push('/dashboard/academix/categories');
+        }
```
Comment (jayasanka-sack) [inline frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:44]: handle non 201 codes. (add else block :) )

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    axios
+      .post(`${window.origin}/core/academix/admin/categories`, values)
+      .then((res: AxiosResponse<Category>) => {
+        if (res.status == 201) {
+          this.setState({ isLoading: false });
+          notification.success({
+            message: 'Success!',
+            description: 'Successfully Created a Category',
+          });
+          this.props.history.push('/dashboard/academix/categories');
```
Comment (jayasanka-sack) [inline frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:43]: Let's redirect to the created category's page. It's hard to identify if there are many categories. (when paginating)

File: frontend/app/src/scenes/Home/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/index.tsx:98]: Is it okay to use a string like `add-item` as the key?

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 2 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:35]: Okay, is there any difference between, `window.location.origin` and `window.origin`?

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 3 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:44]: Working on it!

File: frontend/app/src/scenes/Home/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/index.tsx:98]: Resolved with commit [5c9ed7c](https://github.com/sef-global/sef-core/pull/75/commits/5c9ed7cf928bf44cff036b11cd9abb2e3ccf51e4)

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 2 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx]: Resolved with commit [5c9ed7c](https://github.com/sef-global/sef-core/pull/75/commits/5c9ed7cf928bf44cff036b11cd9abb2e3ccf51e4)

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 3 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:50]: Resolved with commit [5c9ed7c](https://github.com/sef-global/sef-core/pull/75/commits/5c9ed7cf928bf44cff036b11cd9abb2e3ccf51e4)

File: frontend/app/src/scenes/Home/index.tsx (hunk 1 above)
Comment (jayasanka-sack) [reply frontend/app/src/scenes/Home/index.tsx:98]: yes, it's ok. The key is for the item. not for the link

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 3 above)
Comment (jayasanka-sack) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:50]: no just throw an error. the catch block will catch it.

File: frontend/app/src/scenes/Home/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/index.tsx:98]: Got it!

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 3 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:50]: Ah ha, thanks for the guidance! On it!

File: frontend/app/src/scenes/Home/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/index.tsx]: Resolved with commit [8d8d2ca](https://github.com/sef-global/sef-core/pull/75/commits/8d8d2ca1b7aaebafbf573bc0b20e319a0ee04d77)

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx (hunk 3 above)
Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:46]: Resolved with commit [8d8d2ca](https://github.com/sef-global/sef-core/pull/75/commits/8d8d2ca1b7aaebafbf573bc0b20e319a0ee04d77)

File: frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    super(props);
+    this.state = {
+      isLoading: false,
+    };
+  }
+
+  onFinish = (values: any) => {
+    this.setState({ isLoading: true });
+    axios
+      .post(`${window.location.origin}/core/academix/admin/categories`, values)
```
Comment (jayasanka-sack) [inline frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:35]: Let's use + to concatenate these types of strings

Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx:35]: Sure! Is there a reason to use + concatenation here?

Comment (Gravewalker666) [reply frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx]: Resolved with commit [1398e95](https://github.com/sef-global/sef-core/pull/75/commits/1398e959bfe548ee997ef26decd68c7c751f96f7)

**Pull request merged**